src/components/controllers/sensor_controller/sensor_controller.c

```diff
@@ -0,0 +1,220 @@
+// Copyright 2024 MACKI, Krzysztof Gliwinski
+
+#include "sensor_controller.h"
+
+#include <string.h>
+
+#include "adc_wrapper.h"
+#include "macki_log.h"
+#include "sensor_converters.h"
+#include "sensor_driver_definitions.h"
+
+#define TAG "SENSOR_CONTROLLER"
+
+typedef struct {
```

Comment:
you can create static struct here with the sensor_controller_drivers name, and also initialize the values here.
```
static struct {
  ads1115_driver_t* adc_expander;
  lis2dw12_driver_t* accelerometer;
  tmp1075_driver_t* temperature_sensor;
  vl53l0x_driver_t* distance_sensor;
} sensor_controller_drivers = {
    .adc_expander = &adc_expander,
    .accelerometer = &accelerometer,
    .temperature_sensor = &temperature_sensor,
    .distance_sensor = &distance_sensor,
}
```